packages/app/src/specs/RunStatusDots.vue

```diff
@@ -131,10 +142,22 @@ const props = defineProps<{
   specFileExtension: string
 }>()
 
+const emits = defineEmits<{
+  (eventName: 'hasRuns', value: boolean): void
+}>()
+
 const runs = computed(() => {
   return props.gql?.data?.__typename === 'CloudProjectSpec' ? props.gql?.data?.specRuns?.nodes ?? [] : []
 })
 
+const isRunsLoaded = computed(() => {
+  return !!props.gql?.data
+})
+
+watchEffect(() => {
+  emits('hasRuns', isRunsLoaded.value)
```

Comment:
This pattern overall feels like something we could compute at the level above. It might not be doing exactly what you expect:

The `:has-runs="hasRuns"` in `SpecsList.vue` is binding _all rows_ to a single `hasRuns` ref in the scope of `SpecList.vue.` Each `RunStatusDots` component instance will set the value of `hasRuns` when it is rendered, which will affect all rows, making the last `RunStatusDots` that renders the "winner" in what gets set.

I suspect this is not exactly what you intended, and that we want each row to know if _that spec_ has any runs? Or more likely it's enough to know if _any_ spec in the project has runs, which would mean the project is connected and we don't need messaging about connecting it. But that's hard to derive without rendering all specs at once.

Reply:
Removed unused code